### PR TITLE
Disallow unreachable_pub

### PR DESCRIPTION
**Pull request by JonathanPlasse**
Fixed with `cargo clippy --fix` and some manual fixes for IOError

### CODE REVIEW

File: crates/ruff/src/message/text.rs [hunk 1]
```diff
@@ -114,8 +114,8 @@ impl Emitter for TextEmitter {
 }
 
 pub(super) struct RuleCodeAndBody<'a> {
-    pub message_kind: &'a DiagnosticKind,
-    pub show_fix_status: bool,
+    pub(crate) message_kind: &'a DiagnosticKind,
```
Comment (MichaReiser) [inline crates/ruff/src/message/text.rs:117]: Can it be avoided that it adds the pub(crate) if the type itself isn't visible?

Comment (JonathanPlasse) [reply crates/ruff/src/message/text.rs:117]: Currently, it seems not.
I could not find an issue concerning this in rust-lang/rust.

**Comment by JonathanPlasse** [comment]
Should I rebase?

**Comment by JonathanPlasse** [comment]
Just rebased!

**Pull request merged**